mico-core/src/test/java/io/github/ust/mico/core/SwaggerTests.java

```diff
@@ -0,0 +1,69 @@
+package io.github.ust.mico.core;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.hateoas.MediaTypes;
+import org.springframework.http.HttpHeaders;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultMatcher;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static io.github.ust.mico.core.ServiceControllerTests.*;
+import static org.hamcrest.CoreMatchers.endsWith;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.neo4j.internal.kernel.api.IndexQuery.IndexQueryType.exists;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {MicoCoreApplication.class, SwaggerConfig.class})
+@WebAppConfiguration
+@TestPropertySource("classpath:application.properties")
+public class SwaggerTests {
+
+    @Autowired
+    private WebApplicationContext context;
+
+    private MockMvc mvc;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        this.mvc = MockMvcBuilders.webAppContextSetup(this.context).dispatchOptions(true).build();
+    }
+
+    @Test
+    public void vendorExtensionsTest() throws Exception {
+        mvc.perform(get("/v2/api-docs").accept(MediaTypes.HAL_JSON_VALUE))
+            .andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.definitions.Service.properties.name.x-order", notNullValue()))
```

Comment:
to specific for a test/will break without breaking the test beeing considered a bug. Consider not testing this (and alike cases) in favour of manual swagger json review.